webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala

```diff
@@ -217,6 +217,19 @@ class AcceptedNodesLDAPQueryProcessor(
     } }
   }
 
+  override def processOnlyId(query:Query) : Box[Seq[NodeId]] = {
+    //only keep the one of the form Full(...)
+    queryAndChekNodeId(query, NodeInfoService.nodeInfoAttributes, None).map { seq => seq.flatMap {
+      case QueryResult(nodeEntry, _ , _) =>
+        processor.ldapMapper.entryToNode(nodeEntry) match {
+          case Full(node) => Seq(node.id)
```

Comment:
Here, I'm not sure why you need to mapping ? You only want the `NodeId`, and the mapping is not really the goad of that method. Bad nodes would appear elsewhere (generation for ex, or node details, etc). 
I believe you can earn quite a lot more memory here by just usgin `nodeDit.NODES.NODE.idFromDn(nodeEntry.dn)`